.github/workflows/check_security_vulnerability.yml

```diff
@@ -1,27 +1,37 @@
-# This workflow uses actions that are not certified by GitHub.
-# They are provided by a third-party and are governed by
-# separate terms of service, privacy policy, and support
-# documentation.
-
-name: Check Security Vulnerability
+name: Code Quality and Security
 
 on:
   pull_request:
   push:
     branches:
       - main
```

Comment:
Running the job regularly, say once a week, can still be beneficial, even if it is already running on pull_request and pushing to the main branch. Here are some reasons why:

Dependency updates: Regular runs can catch vulnerabilities introduced by dependency updates, which may not be covered by code changes in pull requests or direct pushes.

Environment changes: Changes in the environment or external services may introduce new vulnerabilities or problems that regular runs can help to identify.

Missed changes: If there are changes made directly to the repository that bypass the usual pull request process, a scheduled run can ensure that these changes are also checked.

Reply:
> Dependency updates: Regular runs can catch vulnerabilities introduced by dependency updates, which may not be covered by code changes in pull requests or direct pushes.

Emm, we record the used dependencies in `Cargo.lock` using full version number, they won't change without PR/direct push.

> Missed changes: If there are changes made directly to the repository that bypass the usual pull request process, a scheduled run can ensure that these changes are also checked.


Directly pushing to the main branch is barely used, so every change would be made through a PR.

--------

> Environment changes: Changes in the environment or external services may introduce new vulnerabilities or problems that regular runs can help to identify.

This makes sense to me.